### ISSUE TITLE
Version 2.5.0 - Only guest devices are appearing, home device was removed

### ISSUE THREAD

**Issue by 33racerx** [hjdhjd/homebridge-myq#250]
**Describe The Problem:**

Upon install of 2.5.0 and restart of Homebridge, I can now see devices in which I have guest access, but my home device removed from Homebridge.  Functioning properly in the myQ app.

**To Reproduce:**

I can not reproduce due to my home myQ hub is removed and unable to rediscover.  Tried clearing the homebridge device cache and removing/install the plugin with no change.

**Logs:**

Man Cave Overhead and Scott's Garage are guest access set for me.
**Single Garage Door** is my home door.

```
[1/22/2021, 7:30:20 AM] [Gary Way] myQ API: Successfully connected to the myQ API.
[1/22/2021, 7:30:20 AM] [Gary Way] myQ API: Discovered device family gateway: Hub [Chamberlain Smart Garage Hub] (serial number: xxxxxxx).
[1/22/2021, 7:30:20 AM] [Gary Way] myQ API: Discovered device family garagedoor: Man Cave Overhead (serial number: xxxxxxx, gateway: xxxxxxxx).
[1/22/2021, 7:30:20 AM] [Gary Way] myQ API: Discovered device family garagedoor: Scott’s Garage (serial number: xxxxxxx, gateway: xxxxxxx).
[1/22/2021, 7:30:20 AM] [Gary Way] myQ API: Discovered device family gateway: Scott’s Garage [Liftmaster WiFi GDO AC] (serial number: xxxxxxxx).
[1/22/2021, 7:30:20 AM] [Gary Way] Man Cave Overhead: Adding garagedoor device to HomeKit: Man Cave Overhead (serial number: xxxxxxxxx, gateway: xxxxxxxxxx).
HAP Warning: Characteristic 00000079-0000-1000-8000-0026BB765291 not in required or optional characteristics for service 00000041-0000-1000-8000-0026BB765291. Adding anyway.
[1/22/2021, 7:30:21 AM] [Gary Way] Man Cave Overhead: Door position sensor detected. Enabling battery status support.
[1/22/2021, 7:30:21 AM] [Gary Way] Scott’s Garage: Adding garagedoor device to HomeKit: Scott’s Garage (serial number: xxxxxxx, gateway: xxxxxxxxxx).
[1/22/2021, 7:30:21 AM] [Gary Way] Single Garage Door: Removing myQ device from HomeKit.
[1/22/2021, 7:30:21 AM] [Gary Way] Man Cave Overhead: Closed.
[1/22/2021, 7:30:21 AM] [Gary Way] Scott’s Garage: Closed.
```

**Homebridge Configuration:**

```json
{
            "email": "user@email.com",
            "password": "password",
            "name": "Gary Way",
            "platform": "myQ"
        }
```

**Screenshots:**

**Environment:**

* **Homebridge Version:** v1.1.7
* **Node Version:** v14.15.4
* **Homebridge-myQ Plugin Version**: 2.5.0
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:** iphone 7 ver 14.3
* **Operating System and OS Version:** Ubuntu Focal Fossa (20.04.1 LTS)

**Comment by 33racerx** [comment]
Also, just confirmed, installing previous version 2.4.2 rediscovered my device and removed the guest devices.

**Comment by hjdhjd** [comment]
That behavior is unsurprising and seems about right. In reality they are identical devices, but myQ may (and it seems like they are) treat them as unique. I don't view this as an issue, just a thing to be aware of. Most people shouldn't be constantly changing the credentials they use with this plugin. :smile:

Guest access + HomeKit / `homebridge-myq` should really be for things like Airbnbs that have a guest smart home setup or some such. Or, alternatively, if you want to create a separate plugin-only login for whatever reason.

**Comment by acartola** [comment]
Commenting for two reasons: one, to second this issue - one of my 'guest' garages replaced my own door, and the homekit notifications were unnerving because our door was most definitely shut. Two, to respectfully disagree, and I'll use your own argument against it - my Homekit setup shouldn't change because I went to an AirBnB and got guest access to that garage door. That said, I'm in the same situation where I have permanent guest access to two other doors. If the plugin can't distinguish between them automatically, then we'll need a way to configure it.

**Comment by 33racerx** [comment]
hjdhjd, Thanks for the quick reply.  I can understand your point, but that would assume I changed credentials.

I've been using the same credentials with myQ/Homebridge for quite some time.  Also, the guest access I have is for a friend's garages.  The guest access was sent to me via the myQ app, so it just appears as another home in that App, all under my username/password credentials.  The gateway #'s and serial #'s are unique across all devices.

If more info is needed, let me know, I'm happy to help.  Appreciate the time you put into this plug-in, I use it almost daily!

**Comment by hjdhjd** [comment]
Appreciate your comments @acartola and the note on tone - I welcome the discourse! I know people can get passionate and it’s great to see such interest in all this. With that said, I think we’re talking past each other a bit. As I think I now understand it, here’s the scenario you guys are laying out:

* I have a myQ account for my home garage.
* I go to an Airbnb and get guest access to their garage.
* And when I get access to the Airbnb garage...my original garage disappears in `homebridge-myq`? 

If so...that’s interesting and unintended behavior...good to note as a caveat for folks, but it won’t be something I’ll be prioritizing to further test for the time being. I’ll look at it as time allows, but I’m prioritizing real issues with core functionality that may exist with the new API. Guest access is a nice-to-have feature, and I’ll deal with it more once I’m satisfied we’re in a good place.

The scenario I thought was being outlined originally (but now I think it’s what I just described above) was:

* I have a myQ account for my home garage.
* I go to an Airbnb and use a different email address to get guest access.
* I change user credentials in `homebridge-myq` to the new email address and my old garage doors disappear.

**Comment by hjdhjd** [comment]
> hjdhjd, Thanks for the quick reply. I can understand your point, but that would assume I changed credentials.
> 
> I've been using the same credentials with myQ/Homebridge for quite some time. Also, the guest access I have is for a friend's garages. The guest access was sent to me via the myQ app, so it just appears as another home in that App, all under my username/password credentials. The gateway #'s and serial #'s are unique across all devices.
> 
> If more info is needed, let me know, I'm happy to help. Appreciate the time you put into this plug-in, I use it almost daily!

Yep! See above...I think I misunderstood the original issue.

I will look at this as time allows, but bear with me while we see what other issues come from the API changes. We’ll get there! Appreciate all the effort to test and brave the new world we’re all in now.

**Comment by 33racerx** [comment]
@hjdhjd Thanks!  Version 2.4.2 is working fine for me, I'll watch for updates in the thread.

**Comment by acartola** [comment]
@hjdhjd Just to clarify, I think 33racer and I have our own doors and permanent guest access to someone else’s door. When we update the plugin from 2.4.2 to 2.5.0, *our* doors disappeared from HomeKit and were replaced with one of the guest doors. In my case at least, only one door shows up in my Home, but it isn’t my door - it’s one of the guest ones. I was only using Airbnb to motivate an example of what could happen if someone didn’t have any guest access, and then happened to get one on a trip, based on our experience with preexisting guest access. 

I agree guest doors in HomeKit isn’t a priority and I’m not trying to ask for it at all. If you were to pin me down on what I would like to see ... perhaps the easiest first pass would be to allow a configuration input for the opener serial number (or numbers, for those who have multiple doors on their home), so you don’t have to do the impossible task of figuring out how to distinguish guest doors from an api that doesn’t tell you about it. 

I’m back on 2.4.2 as well but am watching closely & will be happy to help test whenever you get a chance for the next update.

**Comment by hjdhjd** [comment]
> @hjdhjd Just to clarify, I think 33racer and I have our own doors and permanent guest access to someone else’s door. When we update the plugin from 2.4.2 to 2.5.0, *our* doors disappeared from HomeKit and were replaced with one of the guest doors. In my case at least, only one door shows up in my Home, but it isn’t my door - it’s one of the guest ones. I was only using Airbnb to motivate an example of what could happen if someone didn’t have any guest access, and then happened to get one on a trip, based on our experience with preexisting guest access. 
> 
> 
> 
> I agree guest doors in HomeKit isn’t a priority and I’m not trying to ask for it at all. If you were to pin me down on what I would like to see ... perhaps the easiest first pass would be to allow a configuration input for the opener serial number (or numbers, for those who have multiple doors on their home), so you don’t have to do the impossible task of figuring out how to distinguish guest doors from an api that doesn’t tell you about it. 
> 
> 
> 
> I’m back on 2.4.2 as well but am watching closely & will be happy to help test whenever you get a chance for the next update. 

I get it. Thanks for further clarifying. 😀 

That said - the configurability you are looking for already exists - RTFM. 😉 You can already selectively hide openers.

I will spend more time on this issue in time - as I stated above. Downgrade if needed in the meantime. We'll get there.

Thanks.

**Comment by acartola** [comment]
So it does. 

@acartola:
Obtains foot. 
Opens mouth. 
Inserts foot. *shame*

I turned on debug logging to see if I could sort which serial numbers to hide and to show and may have found something, but it’s challenging to piece together the theory on my phone. I’ll look on my desktop later and play around to try and confirm. But a short description of my hunch is that it looks like it enumerates the three accounts correctly, but every update it’s printing out a json that contains two entries, and they’re both pointing to the same account (not mine, one of the guests). I suspect even if I try to hide those serial numbers, I may not get the home I want.

**Comment by hjdhjd** [comment]
I already know what the issue is, but appreciate the amusing followup. :smile: Carry on...

**Comment by acartola** [comment]
Well I was only seeing one appear in the log when the plug-in starts, so I’m going to guess it’s all the same thing then. Carrying on (and thanks again for all your work on this!)

**Comment by hjdhjd** [comment]
Fixed in v2.6.0. Enjoy.

**Comment by acartola** [comment]
Yes, it was fixed in 2.6.0, and still looking fine in 2.6.1. Log indicates all gateways and openers as expected, and the feature options to disable the guest gateways are working swimmingly. Thanks again!

Now one of these days I'm going to have to look into why my model (which has a battery) isn't reporting battery status... I saw the messages in the logs, but it must have been for one of the other openers. 🤷🏻‍♂️
